subsys/shell/shell.c

```diff
@@ -610,7 +620,7 @@ static int execute(const struct shell *shell)
 	}
 
 	/* Parent present means we are in select mode. */
-	if (parent != NULL) {
+	if (IS_ENABLED(CONFIG_SHELL_CMDS_SELECT) && (parent != NULL)) {
```

Comment:
I don't see any code size improvement in this case, because `gcc` already knows that `parent = selected_cmd_get(shell)` is going to be `NULL` when `CONFIG_SHELL_CMDS_SELECT=n`.

Reply:
fixed